src/java/cheshire/prettyprint/CustomPrettyPrinter.java

```diff
@@ -0,0 +1,87 @@
+package cheshire.prettyprint;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.util.DefaultIndenter;
+import com.fasterxml.jackson.core.util.DefaultPrettyPrinter;
+
+import java.io.IOException;
+
+class DynamicIndenter extends DefaultIndenter {
+    public DynamicIndenter(int indentation) {
+        super(new String(new char[indentation]).replace("\0", " "), "\n");
```

Comment:
Also, this seems like a weird way to initialize this, relying on a char array being `\0` and replacing it with zeros, can you explain why you initialize it this way?